tests/test_shorttoken/src/main.xc

```diff
@@ -0,0 +1,69 @@
+// Copyright 2016-2021 XMOS LIMITED.
+// This Software is subject to the terms of the XMOS Public Licence: Version 1.
+#include <xs1.h>
+#include <print.h>
+#include <stdio.h>
+#include "xud.h"
+#include "platform.h"
+#include "shared.h"
+
+#define XUD_EP_COUNT_OUT   5
+#define XUD_EP_COUNT_IN    5
+
+/* Endpoint type tables */
+XUD_EpType epTypeTableOut[XUD_EP_COUNT_OUT] = {XUD_EPTYPE_CTL, XUD_EPTYPE_BUL, XUD_EPTYPE_ISO, XUD_EPTYPE_BUL,XUD_EPTYPE_BUL};
+XUD_EpType epTypeTableIn[XUD_EP_COUNT_IN] =   {XUD_EPTYPE_CTL, XUD_EPTYPE_BUL, XUD_EPTYPE_ISO, XUD_EPTYPE_BUL, XUD_EPTYPE_BUL};
+
+void exit(int);
+
+int TestEp_Bulk(chanend c_out, chanend c_in, int epNum, chanend c_out_0)
+{
+    unsigned int length;
+    XUD_Result_t res;
+
+    XUD_ep ep_out_0 = XUD_InitEp(c_out_0);
+    XUD_ep ep_out = XUD_InitEp(c_out);
+    XUD_ep ep_in  = XUD_InitEp(c_in);
+
+    /* Buffer for Setup data */
+    unsigned char buffer[1024];
+
+    for(int i = 10; i <= 11; i++)
+    {    
+        XUD_GetBuffer(ep_out, buffer, length);
+
+        if(length != i)
+        {
+            printintln(length);
+            TerminateFail(FAIL_RX_LENERROR);
+        }
+
+        if(RxDataCheck(buffer, length, epNum,i))
+        {
+            TerminateFail(FAIL_RX_DATAERROR);
+        }
+
+    }
+
+    XUD_Kill(ep_out_0);
+    exit(0);
+}
+
+
```

Comment:
Quite a few main.xc files `#include "test_main.xc"` and use its common functions.  This file does not.  Is that intentional or an oversight?

Reply:
Some tests cannot use test_main - things that loop back for ever for example. 

Other tests it didn't seem important to run with dummy threads.

Reply:
This test cannot use the func since it requires an extra param - c_sof